Make RunAnywayTarget compatible with python 2.7

Marigold opened the pull request:
## Description
Make `RunAnywayTarget` compatible with python 2.7

## Motivation and Context
`exist_ok` parameter doesn't exist for function `os.makedirs` in python 2.7.12

## Have you tested this? If so, how?
I tested it locally with my existing pipeline (can add unit tests if necessary).

Yoone commented:
Since I am being invited in the review process, it looks good to me!

`os.makedirs()` should indeed raise an `OSError` in both 2.7 and 3.x.

Marigold commented:
> Can you also add a test case?

The original tests were skipped in py27 environment (they were using `exist_ok` and the error was catched & skipped). I fixed them and they now test also the functionality I added (they were failing before the fix). Is there anything else or are we ready to merge this?

Tarrasch commented:
Looks good to me :)
